### PR TITLE
add new date.ts module and argument to getStars function

### PR DESCRIPTION
**Pull request by grmbyrn**
### Changes Made:

1. **Date Formatting Module:**
   - Moved the `formatDate` function into a separate module located at `lib/utils/date.ts`.
   - Implemented in `lib/components/molecules/BlogPostCard.svelte`, `routes/(blog-article)/+layout.svelte` and `routes/rss.xml/+server.ts`
   - The motivation is to centralise the date and time formatting logic and provide a single source of truth for date formatting across all components.
   - By using this helper function, we can easily update the date format in one place, affecting all components simultaneously.
   - This approach facilitates testing of date formatting independently of specific components.

   ```typescript
   // src/lib/utils/date.ts
   import dateFormat from 'dateformat';

   export function formatDate(date: string): string {
     return dateFormat(date, 'UTC:dd/mm/yyyy');
   }

   export function formatTime(date: string): string {
	return dateFormat(date, 'UTC:HH:MM:ss');
   }

2. `getStars` Function Modification:

   - Added an argument to the `getStars` function in the `ProjectCard` component (`lib/components/molecules/ProjectCard.svelte`).
   - The argument (`repoName`) replaces the previous usage of a global variable.
   - This change promotes better code organisation and maintainability by explicitly defining function inputs.
   - The modification improves the function's reusability and makes it more self-contained.

### CODE REVIEW

File: src/lib/utils/date.ts [hunk 1]
```diff
@@ -0,0 +1,9 @@
+import dateFormat from 'dateformat';
+
+export function formatDate(date: string): string {
+	return dateFormat(date, 'UTC:dd/mm/yyyy');
+}
+
+export function formatTime(date: string): string {
```
Comment (josecelano) [inline src/lib/utils/date.ts]: Hi @grmbyrn I think the argument should be renamed to `time`.

**Review by josecelano — CHANGES_REQUESTED** [review]
Hi @grmbyrn It looks good! You only need to rename an argument.

It would be nice in the future to add some unit tests. I would like to know how hard would be to setup unit testing in this project.

For [Nuxt has been a pain](https://github.com/torrust/torrust-index-gui/issues/164).

File: src/lib/utils/date.ts (hunk 1 above)
Comment (grmbyrn) [reply src/lib/utils/date.ts]: Hi @josecelano I made that change to the argument and have started looking into how to add unit tests.

**Comment by josecelano** [comment]
ACK d127462b338713d58fcfffb3561e112b8d6aabdc

**Pull request merged**